packages/portfolio-contract/src/portfolio.exo.ts

```diff
@@ -144,6 +146,62 @@ const eventAbbr = (e: VTransferIBCEvent) => {
   return { destination_channel: dest, sequence };
 };
 
+// Interface definitions for PortfolioKit facets
+const TapI = M.interface('PortfolioTap', {
+  receiveUpcall: M.call(M.record()).returns(M.promise()),
+});
+
+const ParseInboundTransferWatcherI = M.interface('ParseInboundTransferWatcher', {
+  onRejected: M.call(M.any()).returns(),
+  onFulfilled: M.call(M.any()).returns(M.promise()),
+});
+
+const ReaderI = M.interface('PortfolioReader', {
+  getLocalAccount: M.call().returns(M.remotable('LocalAccount')),
+  getStoragePath: M.call().returns(VowShape),
+  getPortfolioId: M.call().returns(M.number()),
+  getGMPInfo: M.call(M.string()).returns(VowShape),
+  getTargetAllocation: M.call().returns(M.or(TargetAllocationShapeExt, M.undefined())),
+});
+
+const ReporterI = M.interface('PortfolioReporter', {
+  publishStatus: M.call().returns(),
+  allocateFlowId: M.call().returns(M.number()),
+  publishFlowSteps: M.call(M.number(), M.arrayOf(M.record())).returns(),
+  publishFlowStatus: M.call(M.number(), M.record()).returns(),
```

Comment:
again, 2nd arg needs to be a dynamic check for `StatusFor['flow']`

That is: it should be a `TypedPattern<StatusFor['flow']>`.